pkg/controller/pod/pod_controller.go

```diff
@@ -198,6 +205,7 @@ func (r *ReconcilePod) Reconcile(request reconcile.Request) (reconcile.Result, e
 		if r.isForNsxNodeAgentPod(request) {
 			reqLogger.Info("Reconciling pod update for network status")
 			r.status.SetNodeConditionFromPod(request.NamespacedName, r.sharedInfo, nil)
+			return reconcile.Result{RequeueAfter: ResyncPeriod}, nil
```

Comment:
If you fix the issue to get node status after operator updates it, do we still need to reconcile in a loop?

Reply:
After the fix, the logic will be:
1. the nsx-node-agent pod is running for less than 180s, temporarily update the network-unavailable to true
2. sleep until 180s
3. finally update the network-unavailable to false
-- in step 2, a common case is that the time is slightly less than 180s, e.g. sleep 160s, but a corner case may be sleep 0.001s. For the corner case, I think we should have a requeue mechanism to double secure it.

In fact, I think the ideal implementation is that we don't invoke more than one update operation in one reconcile, but that needs to refactor for the whole logic,  we cannot complete it in this release.

Reply:
Synced offline and uploaded a new patch set which refactored the logic, replacing goroutine with ResyncAfter